tests/ethhexstrings.nim

```diff
@@ -0,0 +1,151 @@
+type
+  HexQuantityStr* = distinct string
+  HexDataStr* = distinct string
+
+# Hex validation
+
+template stripLeadingZeros(value: string): string =
+  var cidx = 0
+  # ignore the last character so we retain '0' on zero value
+  while cidx < value.len - 1 and value[cidx] == '0':
+    cidx.inc
+  value[cidx .. ^1]
+
+proc encodeQuantity*(value: SomeUnsignedInt): string =
+  var hValue = value.toHex.stripLeadingZeros
+  result = "0x" & hValue
+
+template hasHexHeader*(value: string | HexDataStr | HexQuantityStr): bool =
```

Comment:
This is not a proper check. Please check this template code for `value == ""`.

Reply:
Added :)